helper/resource/testing_new.go

```diff
@@ -442,3 +452,27 @@ func testIDRefresh(ctx context.Context, t testing.T, c TestCase, wd *plugintest.
 
 	return nil
 }
+
+func copyWorkingDir(ctx context.Context, t testing.T, stepNumber int, wd *plugintest.WorkingDir) {
+	if v := os.Getenv(plugintest.EnvTfAccPersistWorkingDir); v != "" {
+		dest := wd.GetHelper().WorkingDirectory() + "_" + strconv.Itoa(stepNumber)
+
+		err := os.Setenv(plugintest.EnvTfAccPersistWorkingDirPath, wd.GetHelper().WorkingDirectory())
```

Comment:
Is this new environment variable just for testing the functionality? I'm wondering if there is a different way we can achieve this. Maybe not for this pull request, but currently we randomly generate where the working directory is located. Instead, maybe the working directory could be passed into the test case (falling back to the random directory), so consumers (and our testing 😄 ) can choose where it goes. e.g.

```go
func TestXXX(t *testing.T) {
  t.Setenv(plugintest.EnvTfAccPersistWorkingDir, "1")
  workDir := t.TempDir()

  resource.TestCase{
    // WorkingDir sets the base directory where testing files used by the testing
    // module. It defaults to a randomized, temporary directory.
    WorkingDir: workDir,
  }

  // Checks against workDir
}
```